runtime/doc/lsp.txt

```diff
@@ -2089,8 +2089,7 @@ connect({host}, {port})                                *vim.lsp.rpc.connect()*
         fun(dispatchers: vim.lsp.rpc.Dispatchers ): vim.lsp.rpc.PublicClient function intended to be passed to |vim.lsp.start_client()| or
         |vim.lsp.start()| on the field cmd
 
-                                         *vim.lsp.rpc.domain_socket_connect()*
-domain_socket_connect({pipe_path})
+connect_pipe({pipe_path})                         *vim.lsp.rpc.connect_pipe()*
```

Comment:
Happy to hear any other suggestions

Reply:
Can this be an overload on `vim.lsp.rpc.connect()` ? That's generally a good pattern and aligns with `sockconnect()`, `serverstart()`, etc.

Reply:
It's possible and I had considered it (see also https://github.com/neovim/neovim/pull/26032/files#r1394275858) but the implementation gets slightly more complex because it adds internal dispatch and I felt the overload would make the method more confusing to use. 

But no strong opinion. Do you have a suggestion how the parameters would look like? Could be a `opts` dict with `pipe`, `host`, and `port` key, or following `sockconnect` with `mode` and `address`

Reply:
I'm not sure why `sockconnect()` first arg is `mode`. The signature could be:

    connect({path_or_host}, {port})

and if `port` is provided it's a pipe, else it's tcp.

Reply:
Turns out that this actually simplifies the implementation too: https://github.com/neovim/neovim/pull/28398/files